### PR TITLE
kube-scheduler: Adjust the structure type to improve the efficiency

### PR DESCRIPTION
**Pull request by zhutong196**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Adjust the type of `PluginToNodeScores map[string]NodeScoreList` to `PluginToNodeScores map[ScorePlugin]NodeScoreList`, In this way, in the `RunScorePlugins` function,` plugintonodescores[pl.name()][index]` can be changed to `nl[index]`, which can improve the calculation efficiency as a whole.

After testing, according to the default number of scoring plug-ins, the time for each node to traverse all scoring plug-ins to get the total score is reduced by about 10us; When the number of cluster nodes is large, the overall optimization process time will be significantly reduced.

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### SIG labels

- sig scheduling

### CODE REVIEW

**Comment by kerthcet** [comment]
FYI we have another PR related https://github.com/kubernetes/kubernetes/pull/108494, but it seems stuck in pending cc @sanposhiho

**Comment by zhutong196** [comment]
> @zhutong196：以下测试**失败**，例如`/retest`重新运行所有失败的测试或`/retest-required`重新运行所有强制失败的测试：
> 
> 测试名称	犯罪	细节	必需的	重新运行命令
> 拉-kubernetes-verify-govet-levee	[483b 费](https://github.com/kubernetes/kubernetes/commit/483bfee511c8dc023f86eaa0fc467ed3f42b27bb)	[关联](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/111729/pull-kubernetes-verify-govet-levee/1556589379059716096)	真的	`/test pull-kubernetes-verify-govet-levee`
> 拉 Kubernetes 单元	[483b 费](https://github.com/kubernetes/kubernetes/commit/483bfee511c8dc023f86eaa0fc467ed3f42b27bb)	[关联](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/111729/pull-kubernetes-unit/1556589378447347712)	真的	`/test pull-kubernetes-unit`
> 拉 Kubernetes 验证	[483b 费](https://github.com/kubernetes/kubernetes/commit/483bfee511c8dc023f86eaa0fc467ed3f42b27bb)	[关联](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/111729/pull-kubernetes-verify/1556589379865022464)	真的	`/test pull-kubernetes-verify`
> [完整的 PR 测试历史](https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=111729)。[您的公关仪表板](https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3Azhutong196)。当您在 PR 中遇到一个未解决的问题时，请通过[链接到](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests)一个[未解决](https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open)的问题来帮助我们减少问题。

/retest

File: pkg/scheduler/framework/runtime/framework.go [hunk 1]
```diff
@@ -902,24 +902,24 @@ func (f *frameworkImpl) RunScorePlugins(ctx context.Context, state *framework.Cy
...
-		pluginToNodeScores[pl.Name()] = make(framework.NodeScoreList, len(nodes))
+		pluginToNodeScores[pl] = make(framework.NodeScoreList, len(nodes))
 	}
 	ctx, cancel := context.WithCancel(ctx)
 	errCh := parallelize.NewErrorChannel()
 
 	// Run Score method for each node in parallel.
 	f.Parallelizer().Until(ctx, len(nodes), func(index int) {
-		for _, pl := range f.scorePlugins {
+		for pl, nl := range pluginToNodeScores {
```
Comment (sanposhiho) [inline pkg/scheduler/framework/runtime/framework.go:914]: This PR violates this constraint. 

https://github.com/kubernetes/kubernetes/blob/a38bb7ed811a18cabe73974afea86f6da34d142b/staging/src/k8s.io/kube-scheduler/config/v1/types.go#L222
```
	// These are called after default plugins and in the same order specified here.
```

**Comment by sanposhiho** [comment]
> FYI we have another PR related https://github.com/kubernetes/kubernetes/pull/108494, but it seems stuck in pending cc @sanposhiho

Yeah, that's a long-lived PR... I need to finalize the implementation asap...

**Comment by dims** [comment]
If you still need this PR then please rebase, if not, please close the PR

**Pull request closed without merging**